File: finalFrontendApplication/src/app/features/employees/controllers/employee-controller.ts
import { Injectable } from "@angular/core";
import { Employee } from "../entities/Employee";
import { Observable } from "rxjs";
import { HttpClient } from "@angular/common/http";




@Injectable({ providedIn: "root" })
export class EmployeeController {

    private url: string = "http://localhost:8080/employees";

     constructor(private http: HttpClient) {}

    getAllEmployees(): Observable<Employee[]> {
        return this.http.get<Employee[]>(this.url);
    }
    

    getAllEmployeesByCompanyId(companyId: string): Observable<Employee[]> {
        return this.http.get<Employee[]>(`${this.url}/company/${companyId}`);
    }

    getEmployeeById(id: string): Observable<Employee> {
        return this.http.get<Employee>(`${this.url}/${id}`);
    }

    createEmployee(employee: Employee): Observable<Employee> {
        return this.http.post<Employee>(this.url, employee);
    }

    updateEmployee(id: string, employee: Employee): Observable<Employee> {
        return this.http.put<Employee>(`${this.url}/${id}`, employee);
    }

    deleteEmployee(id: string): Observable<void> {
        return this.http.delete<void>(`${this.url}/${id}`);
    }



}

File: Android/app/src/main/java/com/bivolaris/myapplication/controller/EmployeeController.java
package com.bivolaris.myapplication.controller;

import com.bivolaris.myapplication.model.CreateEmployeeRequest;
import com.bivolaris.myapplication.model.Employee;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface EmployeeController {

    @GET("employees")
    Call<List<Employee>> getEmployees();

    @GET("employees/{id}")
    Call<Employee> getEmployeeById(@Path("id") String id);

    @GET("employees/company/{id}")
    Call<List<Employee>> getEmployeesByCompanyId(@Path("id") String id);

    @POST("employees")
    Call<Void> createEmployee(@Body CreateEmployeeRequest createEmployeeRequest);

    @PUT("employees/{id}")
    Call<Employee> updateEmployee(@Path("id") String id,@Body Employee employee);

    @DELETE("employees/{id}")
    Call<Void> deleteEmployee(@Path("id") String id);
}
